### DISCUSSION TITLE
Problems with React Portal and Primereact  - with example

### DISCUSSION THREAD

**Discussion by sweco-dkpoju** [primefaces/community#402] [PrimeReact]
When poping out a new browser window by using React createportal I am facing some problems.

1) The splitter doesn't work - you cant move splitbar. I have made an example to illustrate the problem. It seems that mousemove events  are caught on the wrong document - I have made an example here [Example](https://codesandbox.io/s/headless-dawn-hj5pds?file=/src/App.js). In the example I have made my own simple popout but there are npm packages like react-new-window and react-new-improved-window. Those are also tested with the same result.

2) Tooltips on i.e. buttons are shown in wrong window. Seems that the html for tooltip element is placed  in body and by that createportal only handles the button element and not the tooltip element. Workaround is to set the appendto prop to 'self'

**Comment by melloware** [comment]
Your sandbox won't even start for me.

**Reply by sweco-dkpoju** [reply]
Hi,
Thank you for looking into this.
It seems that sandbox in some cases has problems with loading dependencies. I have seen that a couple of time when looking at Primereact examples.
I have attached a video of what is happening - I hope that this will give you an idea about the problem

https://github.com/primefaces/community/assets/56629194/4eb940b6-5c04-488f-84ba-8c53625ff360

**Reply by melloware** [reply]
It is because I had popups blocked.  I have no idea how React Create Portal works but it looks like this popup is truly another HTML window so it looks like JS and CSS is not being loaded into the Portal.

**Reply by melloware** [reply]
Looks like your issue is this: https://github.com/facebook/react/issues/12355

Solution: https://github.com/facebook/react/issues/12355#issuecomment-410996235

```js
const WindowPortal = ({ children }) => {
  const [externalWindow, setExternalWindow] = useState(null);
  const [containerRef, setContainerRef] = useState(null);

  useEffect(() => {
    const containerElement = document.createElement('div');
    const externalWindow = window.open(
      '',
      '',
      'width=600,height=400,left=200,top=200'
    );
    // append the element to the external document before setting ref
    // so that React could detect event bindding correctly
    // https://github.com/facebook/react/issues/12355
    externalWindow.document.body.appendChild(containerElement);
    setContainerRef(containerElement);
    setExternalWindow(externalWindow);

    return () => {
      externalWindow?.close();
    };
  }, []);

  useEffect(() => {
    if (externalWindow) {
      copyStyles(document, externalWindow.document);
    }
  }, [externalWindow]);

  return containerRef && ReactDOM.createPortal(children, containerRef);
};
```

**Reply by sweco-dkpoju** [reply]
@melloware - thank you for your input.

Unfortunately it didn't solve the problem - I have enhanced the example with the information you provided.

For me - it seems that React Portal renders the children in the root and then afterwards  move the elements into the new window element that is provided as parameter to createPortal - but it is only a guess.
After all - it seems not to be a Primereact problem - so I will try to find another forum for this question.

**Comment by mertsincan** [comment]
For now, closed. If the problem persists, please reopen this ticket. Thanks a lot!
